client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/TraceContext.java

```diff
@@ -104,7 +114,18 @@ public void endProcessingNode() {
             trace.setTransaction(getTransaction());
             trace.getNodes().add(node);
 
-            recorder.report(trace);
+            if (checkForSamplingProperties(node)) {
```

Comment:
Although the current set of tests are good for the basic usage, it would be good to add a test scenario as outlined above.